src/Products/Signature/Controllers/SignatureApiController.cs

```diff
@@ -1220,26 +1105,224 @@ public static T LoadXmlData<T>(string xmlPath, string xmlFileName)
+        /// <param name="documentType">string.</param>
+        /// <param name="signaturesData">SignatureDataEntity.</param>
+        /// <param name="signsCollection">SignatureOptionsCollection.</param>
+        private void SignText(string documentType, SignatureDataEntity signaturesData, List<SignOptions> signsCollection)
+        {
+            string xmlPath = this.directoryUtils.DataDirectory.TextDirectory.XmlPath;
+            try
+            {
+                // get xml data of the Text signature
+                string xmlFileName = Path.GetFileNameWithoutExtension(signaturesData.SignatureGuid);
+
+                // Load xml data
+                TextXmlEntity textData = LoadXmlData<TextXmlEntity>(xmlPath, xmlFileName);
+
+                // initiate QRCode signer object
+                TextSigner signer = new TextSigner(textData, signaturesData);
+
+                // prepare signing options and sign document
+                AddSignOptions(documentType, signsCollection, signer);
+            }
+            catch (System.Exception e)
+            {
+                throw new InvalidOperationException(e.Message);
+            }
+        }
+
+        private string GetXmlFilePath(string signatureType, string fileName)
+        {
+            string path;
+            switch (signatureType)
+            {
+                case "stamp":
+                    path = Path.Combine(this.directoryUtils.DataDirectory.StampDirectory.XmlPath, fileName);
+                    break;
+                case "text":
+                    path = Path.Combine(this.directoryUtils.DataDirectory.TextDirectory.XmlPath, fileName);
+                    break;
+                case "qrCode":
+                    path = Path.Combine(this.directoryUtils.DataDirectory.QrCodeDirectory.XmlPath, fileName);
+                    break;
+                case "barCode":
+                    path = Path.Combine(this.directoryUtils.DataDirectory.BarcodeDirectory.XmlPath, fileName);
+                    break;
+                default:
+                    throw new ArgumentNullException("Signature type is not defined");
+            }
+
+            return path;
+        }
+
         /// <summary>
-        /// Save signature data into the XML
+        /// Save signature data into the XML.
         /// </summary>
-        /// <typeparam name="T">Type</typeparam>
-        /// <param name="xmlPath">string</param>
-        /// <param name="xmlFileName">string</param>
-        /// <param name="serializableObject">Object</param>
+        /// <typeparam name="T">Type.</typeparam>
+        /// <param name="xmlPath">XML path.</param>
+        /// <param name="xmlFileName">XML file name.</param>
+        /// <param name="serializableObject">Object.</param>
         private void SaveXmlData<T>(string xmlPath, string xmlFileName, T serializableObject)
         {
-            if (serializableObject == null) { return; }
+            if (serializableObject == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a comparison to 'default(T)' instead or add a constraint to 'T' so that it can't be a value type.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Signature-for-.NET-MVC/pullRequest?prid=5697000)